idaes_examples/notebooks/docs/unit_models/custom_unit_models/liquid_extraction.py

```diff
@@ -0,0 +1,596 @@
+for the aqueous phase,
+**default** - useDefault.
+**Valid values:** {
+**useDefault** - use default package from parent model or flowsheet,
+**PropertyParameterObject** - a PropertyParameterBlock object.}""",
+        ),
+    )
+    CONFIG.declare(
+        "aqueous_property_package_args",
+        ConfigBlock(
+            implicit=True,
+            description="Arguments to use for constructing aqueous phase properties",
+            doc="""A ConfigBlock with arguments to be passed to aqueous phase
+property block(s) and used when constructing these,
+**default** - None.
+**Valid values:** {
+see property package for documentation.}""",
+        ),
+    )
+    CONFIG.declare(
+        "reaction_package",
+        ConfigValue(
+            default=None,
+            domain=is_reaction_parameter_block,
+            description="Reaction package to use for control volume",
+            doc="""Reaction parameter object used to define reaction calculations,
+**default** - None.
+**Valid values:** {
+**None** - no reaction package,
+**ReactionParameterBlock** - a ReactionParameterBlock object.}""",
+        ),
+    )
+    CONFIG.declare(
+        "reaction_package_args",
+        ConfigBlock(
+            implicit=True,
+            description="Arguments to use for constructing reaction packages",
+            doc="""A ConfigBlock with arguments to be passed to a reaction block(s)
+and used when constructing these,
+**default** - None.
+**Valid values:** {
+see reaction package for documentation.}""",
+        ),
+    )
+
+    def build(self):
+        """
+        Begin building model (pre-DAE transformation).
+        Args:
+            None
+        Returns:
+            None
+        """
+        # Call UnitModel.build to setup dynamics
+        super(LiqExtractionData, self).build()
+
+        # Check phase lists match assumptions
+        if self.config.aqueous_property_package.phase_list != ["Aq"]:
+            raise ConfigurationError(
+                f"{self.name} Liquid-Liquid Extractor model requires that the aquoues "
+                f"phase property package have a single phase named 'Aq'"
+            )
+        if self.config.liquid_property_package.phase_list != ["Liq"]:
+            raise ConfigurationError(
+                f"{self.name} Liquid-Liquid Extractor model requires that the liquid "
+                f"phase property package have a single phase named 'Liq'"
+            )
+
+        # Check for at least one common component in component lists
+        if not any(
+            j in self.config.aqueous_property_package.component_list
+            for j in self.config.liquid_property_package.component_list
+        ):
+            raise ConfigurationError(
+                f"{self.name} Liquid-Liquid Extractor model requires that the liquid "
+                f"and aqueous phase property packages have at least one "
+                f"common component."
+            )
+
+        self.liquid_phase = ControlVolume0DBlock(
```

Comment:
I don;t really want to make more work for you, but this looks like exactly what he `MSContactor` model was designed to do: https://github.com/IDAES/idaes-pse/blob/main/idaes/models/unit_models/mscontactor.py

For an example of a solvent extraction model, see: https://github.com/prommis/prommis/tree/main/src/prommis/solvent_extraction

Reply:
This model was developed during my internship (last summer) which might be the reason for this overlap.